deepmerge/tests/test_full.py

```diff
@@ -29,8 +32,12 @@ def test_merge_or_raise_raises_exception():
         "bar": 1,
         "foo": "a string!"
     }
-    with pytest.raises(Exception):
+    with pytest.raises(InvalidMerge, match=r"^no more strategies found for type conflict and arguments \(<deepmerge.merger.Merger object at 0x[0-9a-f]+>, \['foo'\], 0, 'a string!'\), {}$") as exc_info:
```

Comment:
probably no need for the verbose regex. In my experience trying verify things like formatting of error messages ends up being very brittle. 

Using the Exception type to state the category, and verifying that is sufficient.